common/on-chip-calib.h

```diff
@@ -0,0 +1,121 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "notifications.h"
+
+#include <random>
+
+namespace rs2
+{
+    class viewer_model;
+    class subdevice_model;
+    struct subdevice_ui_selection;
+
+    // On-chip Calibration manager owns the background thread
+    // leading the calibration process
+    // It is controlled by autocalib_notification_model UI object
+    // that invokes the process when needed
+    class on_chip_calib_manager : public process_manager
+    {
+    public:
+        on_chip_calib_manager(viewer_model& viewer, std::shared_ptr<subdevice_model> sub,
+            device_model& model, device dev)
+            : process_manager("On-Chip Calibration", model), 
+             _dev(dev), _sub(sub), _viewer(viewer)
+        {
+        }
+
+        // Get health number from the calibration summary
+        float get_health() const { return _health; }
+
+        // Set calibration speed: 0 - slowest, 4 - fastest
+        // On-Chip Calib manager will reduce speed if needed
+        void set_speed(int speed) { _speed = speed; }
```

Comment:
I would also check the input for validity

Reply:
It is not user accessible